librz/include/rz_il/definitions/effect.h

```diff
@@ -0,0 +1,90 @@
+// SPDX-FileCopyrightText: 2021 heersin <teablearcher@gmail.com>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#ifndef RZ_IL_EFFECT_H
+#define RZ_IL_EFFECT_H
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <rz_util.h>
+#include <rz_type.h>
+#include "bitvector.h"
+
+typedef enum {
+	EFFECT_TYPE_NON, // perform none effect will not affect data / control, used for passing info
+	EFFECT_TYPE_DATA,
+	EFFECT_TYPE_CTRL,
+} EFFECT_TYPE;
+
+typedef enum {
+	EFFECT_NOTATION_NON = 0x0,
+	EFFECT_NOTATION_GOTO_SYS = 0x1,
+	EFFECT_NOTATION_GOTO_HOOK = 0x2
+} EFFECT_NOTATION;
+
+typedef enum {
+	EFFECT_LABEL_ADDR,
+	EFFECT_LABEL_SYSCALL,
+	EFFECT_LABEL_HOOK
+	// more
+} EFFECT_LABEL_TYPE;
+
+typedef enum {
+	DATA_EFF_NON,
+	DATA_EFF_ASSIGN,
+	DATA_EFF_INC
+	// maybe more
+} DATA_EFF_OPERATION;
+
+struct effect_label_t {
+	char *label_id; ///< Label name
+	RzILBitVector addr; ///< RzILBitVector address if EFFECT_LABEL_ADDR
+		///< Function pointer if EFFECT_LABEL_SYSCALL / EFFECT_LABEL_HOOK
+	EFFECT_LABEL_TYPE type; ///< type of label
+};
+
+struct control_effect_t {
+	RzILBitVector pc; ///< New Program Counter
+};
+
+struct data_effect_t {
+	const char *var_name; ///< Name of variable, const one
+	int val_index; ///< index to the new value
+	DATA_EFF_OPERATION operation; ///< operation to value and variable
+};
+
+typedef struct control_effect_t *CtrlEffect;
+typedef struct data_effect_t *DataEffect;
+typedef struct effect_label_t *EffectLabel;
+
+typedef struct effect_union_t *Effect;
```

Comment:
Prefix all these with Rz or RzIL. They are in public headers.

Reply:
I think `RzIL` prefix is better in this case.